Test/git-issues/github-issue-3484.dfy.expect

```diff
@@ -0,0 +1,5 @@
+github-issue-3484.dfy(5,2): Error: match has no cases
```

Comment:
> I'm confused by this. The code above reports this as a resolution error. Yet, this test case shows what appears to be 3 verification errors. What am I reading wrong?

The resolution error is only given for match expressions, and L5 has a match statement.

Reply:
> I do suggest rewording the error, since it seems to suggest that a `match` must have cases. It is only a _reachable_ `match` that must have cases. The error message could be improved, perhaps by using the line L11/L17 message and adding a parenthetic remark that somehow communicates "(in fact, this match has no given cases and yet the verifier was unable to prove that the match is unreachable)".

I think conceptually having a match without cases is like a parse error, because it only occurs when the user didn't finish writing the code they intended to compile.

Now, even if there was an actual parse error, you could still have the parser error correct for that, and then run resolution and then run verification, so that the compiler could give feedback on the part of the program that is finished.

Ideally I would let the parser require at least one case for a match, but if there are no cases let it error correct to `case _ => *` (for expressions) and `case _ => {}` for statements. The error message in the case of no cases would be "match has no cases".

I think a user's intuition is that a match without cases is not allowed, so the error message you suggest would be received as:
- "missing case in match: not all possibilities for selector of type int have been covered" -> 
  Is this about my match having no cases?
- "(in fact, this match has no given cases" -> 
  I could see that!!
- "and yet the verifier was unable to prove that the match is unreachable" -> 
  How is that relevant?

---

I will change the current error message to "Match has no cases and this is only allowed when the verifier can prove the match is unreachable." so it doesn't imply something that's incorrect.